lib/livebook/teams/requests.ex

```diff
@@ -304,14 +304,22 @@ defmodule Livebook.Teams.Requests do
   defp add_team_auth(req, nil), do: req
 
   defp add_team_auth(req, team) do
-    token =
-      if team.user_id do
-        "#{team.user_id}:#{team.org_id}:#{team.org_key_id}:#{team.session_token}"
-      else
-        "#{team.session_token}:#{Livebook.Config.agent_name()}:#{team.org_id}:#{team.org_key_id}"
-      end
-
-    Req.Request.merge_options(req, auth: {:bearer, token})
+    if team.offline do
+      Req.Request.append_request_steps(req,
+        unauthorized: fn req ->
+          {req, Req.Response.new(status: 401)}
+        end
+      )
```

Comment:
you can also do:

```suggestion
      Req.merge(adapter: fn req ->
        {req, Req.Response.new(status: 401)}
      end)
```

but it's fine as is!